test/models/repository.test.js

```diff
@@ -597,6 +599,116 @@ describe('Repository', function() {
 
       await assert.isRejected(repo.commit('hmm'), /didirun\.sh did run/);
     });
+
+    describe('recording commit event with metadata', function() {
+      it('reports partial commits', async function() {
+        const workingDirPath = await cloneRepository('three-files');
+        const repo = new Repository(workingDirPath);
+        await repo.getLoadPromise();
+
+        sinon.stub(reporterProxy, 'addEvent');
+
+        // stage only subset of total changes
+        fs.writeFileSync(path.join(workingDirPath, 'a.txt'), 'qux\nfoo\nbar\n', 'utf8');
+        fs.writeFileSync(path.join(workingDirPath, 'b.txt'), 'qux\nfoo\nbar\nbaz\n', 'utf8');
+        await repo.stageFiles(['a.txt']);
+        repo.refresh();
+
+        // unstaged changes remain
+        let unstagedChanges = await repo.getUnstagedChanges();
+        assert.equal(unstagedChanges.length, 1);
+
+        // do partial commit
+        assert.deepEqual(reporterProxy.addEvent.callCount, 0);
+        await repo.commit('Partial commit');
+        assert.deepEqual(reporterProxy.addEvent.callCount, 1);
+        let args = reporterProxy.addEvent.lastCall.args;
+        assert.deepEqual(args[0], 'commit');
+        assert.isTrue(args[1].partial);
+
+        // stage all remaining changes
+        await repo.stageFiles(['b.txt']);
+        repo.refresh();
+        unstagedChanges = await repo.getUnstagedChanges();
+        assert.equal(unstagedChanges.length, 0);
+
+        reporterProxy.addEvent.reset();
+        // do whole commit
+        assert.deepEqual(reporterProxy.addEvent.callCount, 0);
+        await repo.commit('Commit everything');
+        assert.deepEqual(reporterProxy.addEvent.callCount, 1);
+        args = reporterProxy.addEvent.lastCall.args;
+        assert.deepEqual(args[0], 'commit');
+        assert.isFalse(args[1].partial);
+      });
+
+      it.only('reports if the commit was an amend', async function() {
```

Comment:
I was like "why did code coverage decrease so much" and then I was like "ohh." I do the same thing myself at least once a week.